src/HospitalLibrary/Rooms/EquipmentReallocationService.cs

```diff
@@ -0,0 +1,222 @@
+            {
+                snippedInterval.Add(new TimeInterval(start,start=start + duration));
+            }
+            return snippedInterval;
+        }
+
+        public async Task<List<TimeInterval>> GetTakenIntevals(int roomId, DateTime date)
+        {
+            List<TimeInterval> roomTimeIntervals =
+                         await _unitOfWork.AppointmentRepository.GetAllRoomTakenIntervalsForDate(roomId, date);
+            List<TimeInterval> roomReallocationsIntervals =
+                 await _unitOfWork.EquipmentReallocationRepository.GetAllRoomTakenInrevalsForDate(roomId, date);
+
+
+            List<TimeInterval> mixedIntervals = roomReallocationsIntervals.Concat(roomTimeIntervals).ToList();
+            return mixedIntervals = mixedIntervals.OrderBy(x => x.Start).ToList();
+
+        }
+
+        private async Task<List<TimeInterval>> GetAvailableIntervals(List<TimeInterval> takenIntervals, DateTime date, TimeSpan duration)
+        {
+            List<TimeInterval> available = new List<TimeInterval>();
+            DateTime b = date.Date.AddHours(8);
+
+            foreach (TimeInterval interval in takenIntervals)
+            {
+                if (interval.Start.Subtract(b) >= duration)
+                    available.Add(new TimeInterval(b, interval.Start));
+                b = interval.End;
+            }
+            date=date.Date.AddHours(21);
+            if (date.Subtract(b) >= duration)
+                available.Add(new TimeInterval(b,date));
+            return available;
+        }
+
+        private async Task<List<TimeInterval>> getShared(List<TimeInterval> intervalsA, List<TimeInterval> intervalsB ,TimeSpan duration)
+        {
+            List<TimeInterval> shared = new List<TimeInterval>(); 
+            foreach(TimeInterval interval in intervalsA)
+            {
+                foreach(TimeInterval interval2 in intervalsB)
+                {
+                    if (interval.IsOverlaping(interval2)) 
+                    {
+                       var a = makeShared(interval, interval2, duration);
+                        if (a != null)
+                        {
+                            interval.Start = a.End;
+                            shared.Add(a);
+                        }
+                    }
+                }
+            }
+            return shared;
+        }
+
+        private  TimeInterval makeShared(TimeInterval interval, TimeInterval interval2, TimeSpan duration)
+        {
+            if (interval.Start <= interval2.Start && interval.End <= interval2.End && interval.End.Subtract(interval2.Start) >= duration)
+            {
+                return new TimeInterval(interval2.Start, interval.End);
+            }
+            if (interval.Start <= interval2.Start && interval.End >= interval2.End && interval2.End.Subtract(interval2.Start) >= duration)
+            {
+                return new TimeInterval(interval2.Start, interval.End);
+            }
```

Comment:
All if else statements are same, you should join conditions in one method and call it.
if (something) return new TimeInterval(interval2.start, interval.end)
return null

Also time interval already has some of these methods, you should check appointment service and appointment rescheduler